### ISSUE TITLE
Alternative needed ! Software Engineering :Introduction 

### ISSUE THREAD

**Issue by MohamedMandouh** [ossu/computer-science#549]
It seems that the providers of the **Software Development Micromasters** on edX have decided not to make lecture Questions , quizzes  available for audit learners for the 5th course
**Software Engineering :Introduction**
[https://www.edx.org/course/software-engineering-introduction-ubcx-softeng1x](url)

There's another issue : the final exam for the audit learners requires also a verified track accesss.  
( which is very weird :) since there's a separate exam for verified learners) 

Also they only allow a **_one month_**  audit  access to the course materials. 
( in my case the deadline is 28 December) 
It's a huge disappointment for me, since i've been doing this Micromasters for 5 months now. and I'm forced to stop! 
It's the first time I see this policy being applied on edX.

**Comment by hanjiexi** [comment]
Hi @MohamedMandouh, can you try this trick and see if it helps? https://github.com/ossu/computer-science/issues/534

**Comment by MohamedMandouh** [comment]
WOW ! it works !
thank you @hanjiexi 

it seems that those guys just forgot to write some tests :D

**Comment by MohamedMandouh** [comment]
I still find it helpful to think about other options available for software engineering and architecture courses
there's a handful of them available on coursera and edX : 
[https://www.edx.org/course/software-engineering-essentials](url)
[https://www.edx.org/course/software-development-fundamentals-pennx-sd1x](url)
[https://www.coursera.org/specializations/software-development-lifecycle](url)
[https://www.coursera.org/learn/software-architecture](url)

**Comment by hanjiexi** [comment]
Thanks for these suggestions. I'll take a look!

**Comment by MohamedMandouh** [comment]
Hi @hanjiexi  !
I think you may want to take a look at these two courses : 
https://www.edx.org/professional-certificate/agile-development-using-ruby-rails
would they fit somehow in the curriculum? maybe in advanced programming section?

**Comment by waciumawanjohi** [comment]
@MohamedMandouh 
These courses look interesting. But they unfortunately, they aren't a suitable replacement for How to Code, which in the OSS curriculum is the first course after an introductory CS course. The professors write, "Programming proficiency in an object-oriented programming language such as Java, C#, C++, Python, or Ruby is required. We will teach the basics of Ruby at a very accelerated pace that assumes thorough familiarity with OOP inheritance, static/class vs. instance methods and attributes, recursion, hash tables/hash maps, list comprehensions, higher-order functions, lambda expressions. **This course is NOT a good first course in programming.** Basic Unix command-line skills are helpful. Familiarity with Git, GitHub and Heroku will also be helpful." (Emphasis mine)

**Comment by MohamedMandouh** [comment]
@waciumawanjohi 
sorry I made this unclear 
my point is this two courses are good successor to the first four courses of Micromasters where you gain experience with almost all the prerquisites ! 
Software Construction : Java,familiarity with OOP inheritance, static/class vs. instance methods and attributes, recursion, hash tables/hash maps, 
 How to code : list comprehensions, higher-order functions, lambda expressions

I think these two courses are more practical and still focuses on theory .
 "Software Engineering Introduction " is theoritical with almost no code , no assignments  
 so I think they would make a good alternative 
also, the capstone course have been receiving bad reviews which made me think seriously about alternative
see:
 https://www.classcentral.com/course/edx-software-development-capstone-project-8198?fbclid=IwAR1v1MQcMh_kOTIqa4FVWj_p7sIGHjiRO-GX6DuaZhgGcsBJAl9w7AHROxs

**Comment by waciumawanjohi** [comment]
@MohamedMandouh 
Ah! I misunderstood your suggestion, which was clearly expressed. I saw UBCx and immediately jumped to How to Code. Thanks for reiterating your point.

**Comment by spamegg1** [comment]
I am also unable to continue, because all the quizzes are blocked by the new paywall.

I think that
[https://www.edx.org/course/software-engineering-introduction](url)
can be replaced with
[https://www.coursera.org/learn/software-processes](url)

It seems like a perfect replacement and I am dropping the UBCx course and switching to this one.

Maybe
[https://www.edx.org/course/software-development-capstone-project](url)
can be replaced with
[https://www.coursera.org/learn/software-architecture](url)
although this course has UML diagrams and such, which were covered in earlier UBCx courses, but it has a lot of "Capstones" going for it (whatever that means).

There are also other EdX courses in Advanced CS, some replacements might be needed for those as well. What to do about those? When I get there I'll have to go through all this pain again.

This is truly a pitiful state of affairs indeed.

**Comment by waciumawanjohi** [comment]
It is difficult to tell if there is an issue that needs to be addressed. The current state of affairs seems to be:
* There are a number of courses that address the topic of Software Engineering.
* Two favorites are
  * the current recommendation (https://www.edx.org/course/software-engineering-introduction-ubcx-softeng1x)
  * [Software Development Lifecycle Specialization](https://www.coursera.org/specializations/software-development-lifecycle)
* The favorite classes, despite being on different platforms, have paywalled quizzes and feedback.

Unfortunately, paywalled feedback seems to be an increasingly common feature of online courses. If a high quality alternative is found, the course can certainly be changed. At the moment, it seems there is little call to change from the status quo.
